### PR TITLE
Feat: Run prod automation scripts automatically on csv merge

### PR DESCRIPTION
**Pull request by krisrjohnson21**
This PR came from an improvement idea, not an Asana task. If possible, we should try to run the production automation tests automatically every time a change to a `.csv` file is merged to the `main` branch. 

Following [these instructions](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs), I believe these changes will accomplish an automatic run of the production tests, removing the necessity to manually run the GitHub workflow every time a data source is changed.

### CODE REVIEW

**Review by karenhawkinson — COMMENTED** [review]
@krisrjohnson21 What's the expected order of operations? I only upload files to SimpliGov that have been merged to main. If the automated tests run on merge, they would run before the data source is updated in SimpliGov.

Is Confirm PR approved > update SG > merge PR a better process than Merge > update SG > manually run tests?

**Comment by krisrjohnson21** [comment]
> @krisrjohnson21 What's the expected order of operations? I only upload files to SimpliGov that have been merged to main. If the automated tests run on merge, they would run before the data source is updated in SimpliGov.
> 
> Is Confirm PR approved > update SG > merge PR a better process than Merge > update SG > manually run tests?

Good question! I propose that PR approved > update SG > merge PR is better than having to remember to manually run tests. If that's not a consensus, then we can certainly keep the current workflow

**Comment by arkadyan** [comment]
So the benefit of this is that tests run in production automatically, but the tradeoff is that the "manual step" now turns into holding off on merging the PR and remembering to merge it at the correct time? If I'm getting that right, I'd defer to Karen on which workflow she prefers in practice, I don't see an issue with either one.

If we do go with this change, is there a reason not to just run the tests automatically on every change (i.e. remove the `.csv` condition)?

**Comment by arkadyan** [comment]
@krisrjohnson21 I see, thanks for the clarification on the `.csv` files.

**Comment by krisrjohnson21** [comment]
> @krisrjohnson21 I don't feel strongly as long the process is clearly defined and documented.

Good call. I've added a [Notion page](https://www.notion.so/mbta-downtown-crossing/QA-for-Production-Data-Source-Updates-92f66602e0444bf49cb69718ede30218) with documentation. Grateful for any feedback!

**Review by arkadyan — APPROVED** [review]
👍

**Pull request merged**
